components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -688,15 +682,42 @@ private static ByteArray generateRandom() {
         return new ByteArray(bytes);
     }
 
-    private StartRegistrationOptions buildStartRegistrationOptions(User user) {
+    private StartRegistrationOptions buildStartRegistrationOptions(User user, Boolean requireResidentKey) {
 
         return StartRegistrationOptions.builder()
-                .user(buildUserIdentity(user)).build();
+                .user(buildUserIdentity(user))
+                .timeout(Integer.parseInt(timeout))
+                .authenticatorSelection(buildAuthenticatorSelection(requireResidentKey))
+                .extensions(RegistrationExtensionInputs.builder().build())
+                .build();
+    }
+
+    private AuthenticatorSelectionCriteria buildAuthenticatorSelection(Boolean requireResidentKey) {
```

Comment:
Change to use the primitive data type here

Reply:
Addressed with https://github.com/wso2-extensions/identity-local-auth-fido/pull/58/commits/b5d30d70b50ddc46d7d07c2043fc0f0d876d94a2